akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/javadsl/ReplicationSettings.scala

```diff
@@ -0,0 +1,229 @@
+
+  def withOtherReplicas(otherReplicas: JSet[Replica]): ReplicationSettings[Command] =
+    copy(otherReplicas = otherReplicas)
+
+  /**
+   * Set the timeout for events being completely processed after arriving to a node in the replication stream
+   */
+  def withEntityEventReplicationTimeout(duration: Duration): ReplicationSettings[Command] =
+    copy(entityEventReplicationTimeout = duration)
+
+  /**
+   * Run up to this many parallel updates over sharding. Note however that updates for the same persistence id
+   * is always sequential.
+   */
+  def withParallelUpdates(parallelUpdates: Int): ReplicationSettings[Command] =
+    copy(parallelUpdates = parallelUpdates)
+
+  /**
+   * Change projection provider
+   */
+  def withProjectionProvider(projectionProvider: ReplicationProjectionProvider): ReplicationSettings[Command] =
+    copy(projectionProvider = projectionProvider)
+
+  /**
+   * Add an interceptor to the gRPC event producer for example for authentication of incoming requests
+   */
+  def withEventProducerInterceptor(interceptor: EventProducerInterceptor): ReplicationSettings[Command] =
+    copy(eventProducerInterceptor = Optional.of(interceptor))
+
+  /**
+   * Allows for changing the settings of the replicated entity, such as stop message, passivation strategy etc.
+   */
+  def configureEntity(
+      configure: java.util.function.Function[
+        Entity[Command, ShardingEnvelope[Command]],
+        Entity[Command, ShardingEnvelope[Command]]]): ReplicationSettings[Command] =
+    copy(configureEntity = configure)
+
+  private def copy(
+      selfReplicaId: ReplicaId = selfReplicaId,
+      entityTypeKey: EntityTypeKey[Command] = entityTypeKey,
+      eventProducerSettings: EventProducerSettings = eventProducerSettings,
+      streamId: String = streamId,
+      otherReplicas: JSet[Replica] = otherReplicas,
+      entityEventReplicationTimeout: Duration = entityEventReplicationTimeout,
+      parallelUpdates: Int = parallelUpdates,
+      projectionProvider: ReplicationProjectionProvider = replicationProjectionProvider,
+      eventProducerInterceptor: Optional[EventProducerInterceptor] = eventProducerInterceptor,
+      configureEntity: java.util.function.Function[
+        Entity[Command, ShardingEnvelope[Command]],
+        Entity[Command, ShardingEnvelope[Command]]] = configureEntity): ReplicationSettings[Command] =
+    new ReplicationSettings[Command](
+      selfReplicaId,
+      entityTypeKey,
+      eventProducerSettings,
+      streamId,
+      otherReplicas,
+      entityEventReplicationTimeout,
+      parallelUpdates,
+      projectionProvider,
+      eventProducerInterceptor,
+      configureEntity)
+
+  override def toString =
+    s"ReplicationSettings($selfReplicaId, $entityTypeKey, $streamId, ${otherReplicas.asScala.mkString(", ")})"
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def toScala: SReplicationSettings[Command] =
+    SReplicationSettings[Command](
```

Comment:
Yeah, it can't be converted because we can't transform entity back and forth, instead it is manually applied in Java DSL `Replication.grpcReplication`